common/meta/types/src/user_info.rs

```diff
@@ -94,17 +94,49 @@ impl TryFrom<Vec<u8>> for UserInfo {
 #[serde(default)]
 pub struct UserOption {
     flags: BitFlags<UserOptionFlag>,
+
+    default_role: Option<String>,
 }
 
 impl UserOption {
     pub fn new(flags: BitFlags<UserOptionFlag>) -> Self {
-        Self { flags }
+        Self {
+            flags,
+            default_role: None,
+        }
+    }
+
+    pub fn empty() -> Self {
+        Default::default()
+    }
+
+    pub fn with_flags(mut self, flags: BitFlags<UserOptionFlag>) -> Self {
+        self.flags = flags;
+        self
+    }
+
+    pub fn with_default_role(mut self, default_role: Option<String>) -> Self {
+        self.default_role = default_role;
+        self
+    }
+
+    pub fn with_set_flag(mut self, flag: UserOptionFlag) -> Self {
+        self.flags.insert(flag);
+        self
     }
 
     pub fn flags(&self) -> &BitFlags<UserOptionFlag> {
         &self.flags
     }
 
+    pub fn default_role(&self) -> &Option<String> {
```

Comment:
```suggestion
    pub fn default_role(&self) -> Option<&str> {
```

Reply:
thank you, fixed in https://github.com/datafuselabs/databend/pull/6847/commits/c9a8ac951ac36bfe54358f5620d31f1c12527cde